### PR TITLE
[Java] Static Modifier on Internal Interface pmd #1356

### PR DESCRIPTION
**Pull request by vishva007**
Please, prefix the PR title with the language it applies to within brackets, such as *[java]* or *[apex]*. If not specific to a language, you can use *[core]*

Before submitting a PR, please check that:
 - [ ] The PR is submitted against `master`. The PMD team will merge back to support branches as needed.
 - [ ] `./mvnw test` passes.
 - [ ] `./mvnw pmd:check` passes.
 - [ ] `./mvnw checkstyle:check` passes. [Check this for more info](https://github.com/pmd/pmd/blob/master/CONTRIBUTING.md#code-style)

**PR Description:**
Issue #1356

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">This changeset introduces 58 new violations and 0 new errors,<br />
removes 58 violations and 2 errors. <a href="http://chunk.io/pmd/d0331024ab3c4f81b5aee09560fd6f0d/diff/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by vishva007** [comment]
Hi, 

Should the issues raised from the pmd-test be fixed in this pull request itself ? I think, those new warning came because of change in the warning text.

The two errors seem to come from test files, so are those things supposed to be fixed ? Will look into those if those need a fix.

**Comment by adangel** [comment]
This is merged now and will be part of PMD 6.9.0. Thanks for your contribution!

File: pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/UnnecessaryModifier.xml [hunk 1]
```diff
@@ -673,6 +673,20 @@ enum Foo {
...
         name = s;
     }
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>Static Modifier on interface</description>
+        <expected-problems>1</expected-problems>
+        <expected-messages>
+            <message>Unnecessary modifier 'static' on interface 'Bar': types nested within an interface type are implicitly static</message>
```
Comment (oowekyala) [inline pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/UnnecessaryModifier.xml]: The explanation is now wrong: it should read something like `member interfaces are implicitly static`

Comment (jsotuyod) [reply pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/UnnecessaryModifier.xml]: it actually applies to all nested types, so I dont' think this is wrong at all..

```
public interface Foo {
  public class Bar { } // static here would be redundant
}
```

**Pull request merged**